### PR TITLE
Let constructor accept constructed instance

### PR DESCRIPTION
**Pull request by enricomarino**
Let it accept itself... :)

```js
const cid1 = new CID(...)
const cid2 = new CID(cid1)
cid1.equals(cid2) // true
cid1 === cid2 // false
```

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -202,4 +202,20 @@ describe('CID', () => {
...
+    const cid2 = new CID(cid1)
+
+    it('constructor accept constructed instance', () => {
+      expect(
+        cid1.equals(cid2)
+      ).to.be.eql(true)
+
+      expect(
+        cid1 !== cid2
+      ).to.be.eql(true)
```
Comment (diasdavid) [inline test/index.spec.js]: What about:

`expect(cid1).to.not.equal(cid2)`

File: test/index.spec.js [hunk 2]
```diff
@@ -202,4 +202,20 @@ describe('CID', () => {
...
+
+  describe('idempotence', () => {
+    const h1 = 'QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n'
+    const cid1 = new CID(h1)
+    const cid2 = new CID(cid1)
+
+    it('constructor accept constructed instance', () => {
+      expect(
+        cid1.equals(cid2)
+      ).to.be.eql(true)
```
Comment (diasdavid) [inline test/index.spec.js]: `expect(cid1.equals(cid2)).to.equal(true)`

In mocha, eql and equal are different:
- `eql` - deep object equal
- `equal` - value equal

Comment (enricomarino) [inline test/index.spec.js]: Thanks for the explanation! You're right! For primitive value `equal` is the way.

Comment (enricomarino) [inline test/index.spec.js]: Maybe we should revise the boolean assertions in the preceding tests (from which I started to be consistent)

File: test/index.spec.js (hunk 1 above)
Comment (enricomarino) [inline test/index.spec.js]: more concise, better :)
even thought I'd like to keep separate the assertion to test from the expected value
```js
expect(cid1 !== cid2).to.equal(true)
```
or
```js
expect(cid1 === cid2).to.equal(false)
```

Comment (enricomarino) [inline test/index.spec.js]: what about:

```
expect(...).to.be.true
```

for boolean assertions

File: test/index.spec.js (hunk 2 above)
Comment (enricomarino) [inline test/index.spec.js]: what about the followings

`expect(...).to.be.true`
`expect(...).to.be.false`

for boolean assertions

**Pull request merged**